events/datadog.go

```diff
@@ -143,32 +130,41 @@ func (e *DataDog) NewBatchReapableEvent(rs []Reapable) error {
 			triggering = append(triggering, r)
 		}
 	}
+	// no events triggering
 	if len(triggering) == 0 {
 		return nil
 	}
+
 	log.Info("Sending batch DataDog events for %d reapables.", len(triggering))
-	// j keeps track of which reapable
-	j := 0
-	for j < len(triggering) {
+	// this is a bin packing problem
+	// we ignore its complexity because we don't care (that much)
+	for j := 0; j < len(triggering); {
+		var written int64
 		buffer := *bytes.NewBuffer(nil)
-		// i keeps track of how many reapables
-		// have been written to a buffer
-		for j < len(triggering) {
-			buffer.ReadFrom(triggering[j].ReapableEventTextShort())
-			buffer.WriteString("\n")
-
-			// when we've written 3 reapables
-			// move on to the next buffer
-			if (j%2 == 0 && j != 0) || j == len(triggering)-1 {
-				// send events in this buffer
-				err := e.NewEvent("Reapable resources discovered", buffer.String(), nil, nil)
+		for moveOn := false; j < len(triggering) && !moveOn; {
+			size := int64(triggering[j].ReapableEventTextShort().Len())
```

Comment:
This logic looks good. You should perhaps put `triggering[j].ReapableEventTextShort()` into a variable since it's used twice.